libfreerdp/core/multitransport.c

```diff
@@ -17,35 +17,155 @@
  * limitations under the License.
  */
 
+#include <winpr/assert.h>
 #include <freerdp/config.h>
+#include <freerdp/log.h>
 
+#include "rdp.h"
 #include "multitransport.h"
 
-#include <freerdp/log.h>
-
 #define TAG FREERDP_TAG("core.multitransport")
 
-int rdp_recv_multitransport_packet(rdpRdp* rdp, wStream* s)
+struct rdp_multitransport
+{
+	rdpRdp* rdp;
+
+	UINT32 requestId;         /* requestId (4 bytes) */
+	UINT16 requestedProtocol; /* requestedProtocol (2 bytes) */
+	UINT16 reserved;          /* reserved (2 bytes) */
```

Comment:
We can avoid storing reserved that is not used at all.

Reply:
true, was just for completeness